rp235x-hal-examples/src/bin/rom_funcs.rs

```diff
@@ -234,31 +235,58 @@ fn get_sys_info_chip_info<T>(uart: &mut T)
 where
     T: core::fmt::Write,
 {
-    let mut buffer = [0u32; 16];
-    let result = unsafe { hal::rom_data::get_sys_info(buffer.as_mut_ptr(), buffer.len(), 0x0001) };
-    _ = writeln!(uart, "get_sys_info(CHIP_INFO/0x0001) -> {}", result);
-    _ = writeln!(uart, "\tSupported Flags: {:#06x}", buffer[0]);
-    _ = writeln!(uart, "\tRP2350 Package ID: {:#010x}", buffer[1]);
-    _ = writeln!(uart, "\tRP2350 Device ID : {:#010x}", buffer[2]);
-    _ = writeln!(uart, "\tRP2350 Wafer ID  : {:#010x}", buffer[3]);
+    let result = hal::rom_data::sys_info_api::chip_info();
```

Comment:
These nested matches can be collapsed.

```rust
let chip_info = match hal::rom_data::sys_info_api::chip_info() {
    Ok(Some(info)) => info,
    Ok(None) => {
        _ = writeln!(uart, "chip_info() not supported");
        return;
    }
    Err(e) => {
        _ = writeln!(uart, "Failed to get chip info : {:?}", e);
        return;
    }
};
```

Are we expecting chips to not support chip info?